### ISSUE TITLE
React 18 Upgrade

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/web-stories-wp#10945]
## Feature Description

React 18 is now [available as a release candidate](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html)

The biggest changes relevant for us that I see so far:

* Replacing `ReactDOM.render()` with `createRoot()`
* Automatic batching
  We can probably remove `useBatchingCallback` from `packages/react`
* `useId()` -> new hook for generating unique IDs on both the client and server
  We obviously don't use SSR, but maybe we can replace `uuidv4` in some places with that.
* `globalThis.IS_REACT_ACT_ENVIRONMENT`
  Probably no change needed besides updating Testing Library
* No more "setState on unmounted component" warnings
  We added some workarounds in the past to prevent this warnings (e.g. `let mounted` or `isMounted = useRef()`). Should be able to remove these now.

Some other noteworthy APIs:

* `useSyncExternalStore`
* `startTransition`
* `useDeferredValue`

**Comment by swissspidy** [comment]
From @sayedtaqui:

If I upgrade our react version to 18, the editor works with a warning of "`ReactDOM.render` is no longer supported" but then we have three more dependencies which do not accept major upgrade like us ( using `^` instead of `>=` version notation ), so the editor breaks when we interact with the components using those dependencies ( `react-virtual`, `react-modal`, `@wmik/use-media-recorder` ) 

In other words, we need to wait for these dependencies to declare React 18 support or just contribute changes ourselves for doing so.

* https://github.com/wmik/use-media-recorder
* https://github.com/TanStack/react-virtual
* https://github.com/reactjs/react-modal

**Comment by swissspidy** [comment]
Created some PRs:

* https://github.com/reactjs/react-modal/pull/938
* https://github.com/wmik/use-media-recorder/pull/20
* https://github.com/TanStack/react-virtual/pull/278

**Comment by swissspidy** [comment]
All PRs have been merged, now we just have to wait for new releases :-)

**Comment by timarney** [comment]
Something that might need some testing is the "double useEffect" call when in dev mode in strict mode.  

https://www.youtube.com/watch?v=j8s01ThR7bQ

Will probably result in some double fetch calls etc... not sure what else.

**Comment by swissspidy** [comment]
Yup, it's a new behavior in Strict Mode (only when in development mode).

There are a few places where we do this, for example on the settings page (fetching fonts, settings, publisher logos on mount)
